packages/pyright-internal/src/analyzer/sourceFile.ts

```diff
@@ -135,6 +135,18 @@ class WriteableData {
     builtinsImport: ImportResult | undefined;
     // True if the file appears to have been deleted.
     isFileDeleted = false;
+
+    dump() {
+        return JSON.stringify(this, (key, value) => {
```

Comment:
I think this may be unsafe. There are data structures in the WritableData that have circularities, so JSON.stringify will fail. I recommend outputting specific fields that you're interested in. Also, maybe pick a better name than `dump`.

Reply:
Sure will do. The replacer should have prevented the circular printing but that would be safer.